### PR TITLE
Perform GET Request on recents endpoint and create all the required models

### PR DESCRIPTION
**Pull request by nyjalusc**
* Created two models BoxRecentItem and BoxIteratorRecentItems
* Created API class and request class to perform the operation

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @nyjalusc has signed the CLA. Thanks for the pull request!

File: box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiRecentItems.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        super(session);
+    }
+
+    /**
+     * Gets the URL for getting the recent items
+     *
+     * @return the URL string for getting the recent items
+     */
+    protected String getRecentItemsUrl() {
+        return String.format("%s/recent_items", getBaseUri());
```
Comment (rgcorrea) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiRecentItems.java]: Can we put constants into "private static final String" variables ?

File: box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxRecentItem.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * A class representing Box recent item
+ */
+public class BoxRecentItem extends BoxJsonObject {
+    private static final long serialVersionUID = -2642748896882484887L;
+
+    public static final String TYPE = "recent_item";
+
+    public static final String FIELD_INTERACTION_TYPE = "interaction_type";
```
Comment (rgcorrea) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxRecentItem.java]: Should these constants be public ?
I believe whoever parses need to have access, but not sure.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiRecentItems.java (hunk 1 above)
Comment (nyjalusc) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiRecentItems.java]: +1

File: box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxRecentItem.java (hunk 2 above)
Comment (nyjalusc) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxRecentItem.java]: I found a model "BoxDownload", which I used as reference. It has private accessibility for static properties, so I believe it should be fine to narrow it down.

**Comment by doncung** [comment]
+1

**Pull request merged**